core/src/main/scala/org/apache/spark/util/collection/BitSet.scala

```diff
@@ -58,6 +58,20 @@ class BitSet(numBits: Int) extends Serializable {
   }
 
   /**
+   * Clear all the bits up to a given index
+   */
+  def clearUntil(bitIndex: Int) {
+    val wordIndex = bitIndex >> 6 // divide by 64
+    var i = 0
+    while(i < wordIndex) { words(i) = 0; i += 1 }
```

Comment:
java.util.Arrays.fill can do this in one line. It won't be faster but could be a tiny bit cleaner. Up to your taste whether you want to change this and other occurrences in the file.